internal/vendors/accessor.go

```diff
@@ -144,14 +144,11 @@ func (p *postgresVendorAccessor) GetByLocation(ctx context.Context, location str
 	return vendors, nil
 }
 
-func (p *postgresVendorAccessor) GetByProduct(ctx context.Context, product string) ([]Vendor, error) {
-	// Split the product string into words
-	words := strings.Fields(product)
-
+func (p *postgresVendorAccessor) GetByProductWords(ctx context.Context, productWords []string) ([]Vendor, error) {
```

Comment:
Won't it be better changing the name to getByProductDescription? Because the param that you provide is used for description filtering. 
I just think as a client, seeing the required args as `productWords` doesnt describe what I need to provide

Reply:
Should I use `GetByProductDescription` and the parameter is still list of words, or you're suggesting to revert it back to the original that we split the description inside the accessor function?

Reply:
My suggestion is 

```func (p *postgresVendorAccessor) GetByProductDescription(ctx context.Context, productDescription string) ([]Vendor, error) { ```

Same with what Valerian said before

Reply:
If we change it like that again, then the splitting logic will be in the accessor like the original state. Is that okay?

cc: @valeelim

Reply:
I think what @valenciusap17 means is that `productDescription` should be a list of strings, i.e.
`func (p *postgresVendorAccessor) GetByProductDescription(ctx context.Context, productDescription []string) ([]Vendor, error) { `

The issue here is just the function/param naming of `Words`, as someone who reads the code, I can't really tell what the function wants/does. What do "words" in this case represent?